src/app/dialogs/change-password-dialog/change-password-dialog.component.ts

```diff
@@ -0,0 +1,43 @@
+import { Component, OnInit } from '@angular/core';
+import { AbstractControl, FormBuilder, FormGroup, ValidationErrors, Validators } from '@angular/forms';
+
+@Component({
+  selector: 'app-change-password-dialog',
+  templateUrl: './change-password-dialog.component.html',
+  styleUrls: ['./change-password-dialog.component.scss', '../edit-profile-dialog/edit-profile-dialog.component.scss']
+})
+export class ChangePasswordDialogComponent implements OnInit {
+  changePasswordForm!: FormGroup;
+
+  private readonly passwordRegex: RegExp = /^(?=\D*\d)(?=[^a-z]*[a-z])(?=[^A-Z]*[A-Z]).{8,30}$/;
+
+  constructor(private fb: FormBuilder) { }
+
+  ngOnInit(): void {
+    this.changePasswordForm = this.fb.group({
+      oldPassword: ['', Validators.required],
+      newPassword: ['', [Validators.required, this.passwordValidator.bind(this)]],
+      confirmPassword: ['', [Validators.required, this.confirmPasswordValidator.bind(this)]]
+    })
```

Comment:
it is better to provide only function calls inside ngOnInit

so you can create `private` function `setForm(): void` and move form assigning there. and in ngOnInit provide simple `this.setForm()` call